### PR TITLE
SIDM-6606--1 - Added ability to delete user based on roles

### PR DESCRIPTION
**Pull request by Thomas-Geraghty**
### JIRA link ###

https://tools.hmcts.net/jira/browse/SIDM-6606

### Change description ###

- Added new API method 'getAllRoles' which utilises systemAxios (axis instance auth'd with system user credentials) to get all roles within IDAM
- Added UserDeleteController that will handle deleting of user, validating fields etc.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by Thomas-Geraghty** [comment]
Lets get https://github.com/hmcts/idam-user-dashboard/pull/239 merged, then we can fix the async test actions for this PR

**Comment by Thomas-Geraghty** [comment]
This is ready for review now @KianKwa

File: src/test/functional/delete-user-test.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import {config as testConfig} from '../config';
+import * as Assert from 'assert';
+import {randomData} from './shared/random-data';
+import {BETA_FEATURES} from '../../main/app/feature-flags/flags';
+
+Feature('Delete User');
+
+const PARENT_ROLE = testConfig.TEST_SUITE_PREFIX + randomData.getRandomString(10);
+const ASSIGNABLE_CHILD_ROLE = testConfig.TEST_SUITE_PREFIX + randomData.getRandomString(10);
+const INDEPENDANT_CHILD_ROLE = testConfig.TEST_SUITE_PREFIX + randomData.getRandomString(10);
```
Comment (KianKwa) [inline src/test/functional/delete-user-test.ts:13]: Probably something small. random.getRandomString(0 is called a few times here and some other classes too. Might be better to create a utility method called createData() or something like that?

File: src/test/functional/delete-user-test.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+BeforeSuite(async () => {
+  await createAssignableRoles(PARENT_ROLE);
+  await createAssignableRoles(ASSIGNABLE_CHILD_ROLE);
+  await createAssignableRoles(INDEPENDANT_CHILD_ROLE);
+  // Assigning self role with the child role so the this user can also delete same level users
+  await assignRolesToParentRole(PARENT_ROLE, [ASSIGNABLE_CHILD_ROLE, PARENT_ROLE]);
+  await createUserWithRoles(PARENT_ROLE_EMAIL, testConfig.PASSWORD, testConfig.USER_FIRSTNAME, [testConfig.RBAC.access, PARENT_ROLE]);
+});
+
+Scenario('I as a user should not be able delete user if I do not have the role with right to delete',
```
Comment (KianKwa) [inline src/test/functional/delete-user-test.ts:25]: "should not be able delete"...grammatical error here. Same below.

**Review by KianKwa — APPROVED** [review]

File: src/test/functional/delete-user-test.ts (hunk 1 above)
Comment (Thomas-Geraghty) [reply src/test/functional/delete-user-test.ts:13]: @ashwini-mv, we can address this in a future PR? :)

**Pull request merged**